src/ripple/protocol/Rules.h

```diff
@@ -90,5 +96,31 @@ class Rules
     operator!=(Rules const& other) const;
 };
 
+std::optional<Rules> const&
+getCurrentTransactionRules();
+
+void
+setCurrentTransactionRules(std::optional<Rules> r);
+
+/** RAII class to set and restore the current transaction rules
+ */
+class CurrentTransactionRulesGuard
+{
+public:
+    explicit CurrentTransactionRulesGuard(Rules r)
```

Comment:
This class should not be CopyConstructible nor CopyAssignable.

Reply:
Yep, agreed, thanks! Fixed in `94bbc9c8a9 [fold] CurrentTransactionRulesGuard should not be copyable or assignable`